lib/MusicBrainz/Server/Entity/URL.pm

```diff
@@ -1,42 +1,72 @@
 package MusicBrainz::Server::Entity::URL;
 use Moose;
 
-use Encode 'decode';
 use MooseX::Types::URI qw( Uri );
-use MusicBrainz::Server::Filters;
-use URI::Escape;
-use Try::Tiny;
 
 extends 'MusicBrainz::Server::Entity::CoreEntity';
 with 'MusicBrainz::Server::Entity::Role::Linkable';
 
 has 'url' => (
-    is => 'rw',
+    is => 'ro',
```

Comment:
The attribute was never set except on construction, so this doesn’t really change anything. However, because I added a number of lazy attributes that cache values computed from the `url` attribute here, I thought I should mark it as constant.